error-prone-contrib/pom.xml

```diff
@@ -62,6 +62,13 @@
             <artifactId>auto-value-annotations</artifactId>
             <scope>provided</scope>
         </dependency>
+        <!-- XXX: Drop this dependency declaration once NullAway in JSpecify
+        mode no longer requires it when building with JDK 23+. -->
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <scope>provided</scope>
+        </dependency>
```

Comment:
Without this version declaration builds with JDK 23+ fail with:
```
[WARNING] Cannot find annotation method 'when()' in type 'javax.annotation.Nonnull': class file for javax.annotation.Nonnull not found
[WARNING] /home/sschroevers/workspace/picnic/error-prone-support/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java: unknown enum constant javax.annotation.meta.When.MAYBE
  reason: class file for javax.annotation.meta.When not found
```

(This causes a failure because we build with `-Werror`.)

It's rather ironic that enabling JSpecify mode forces one to add a "competing" dependency to the classpath. :upside_down_face:

Reply:
Hrm, this is a weird one!  Particularly that it's tied to JSpecify mode, I have no idea what's going on.  Can you file an issue for this one?

Reply:
Sure! I'll first try to narrow down which aspect of `ReactorRules` triggers the issue.

Reply:
Alright, filed uber/NullAway#1171.